app/models/incident.rb

```diff
@@ -7,7 +7,17 @@ class Incident < ApplicationRecord
 
   accepts_nested_attributes_for :incident_competitions, allow_destroy: true
 
-  scope :resolved, -> { where.not(resolved_at: nil) }
+  attribute :visibility, :integer, default: :visible
+
+  enum visibility: {
+    draft: 0,          # Only visible to WRC members
+    staff: 1,          # Visible to WCA staff members only
+    visible: 2,        # Visible to everyone
+  }
```

Comment:
If the `enum`ized column is an integer, then you can define the enum values as an array
```suggestion
  enum visibility: [
    :draft,          # Only visible to WRC members
    :staff,          # Visible to WCA staff members only
    :visible,        # Visible to everyone
  ]
```